src/tigerbeetle-java/src/main/java/com/tigerbeetle/Accounts.java

```diff
@@ -0,0 +1,317 @@
+package com.tigerbeetle;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A {@link Batch batch} of accounts.
+ */
+public final class Accounts extends Batch {
+
+    interface Struct {
+
+        public static final int Id = 0;
+        public static final int UserData = 16;
+        public static final int Reserved = 32;
+        public static final int Ledger = 80;
+        public static final int Code = 84;
+        public static final int Flags = 86;
+        public static final int DebitsPending = 88;
+        public static final int DebitsPosted = 96;
+        public static final int CreditsPending = 104;
+        public static final int CreditsPosted = 112;
+        public static final int Timestamp = 120;
+
+        public static final int SIZE = 128;
+    }
+
+    static final Accounts EMPTY = new Accounts(0);
+
+    /**
+     * Constructs an empty batch of accounts with the desired maximum capacity.
+     * <p>
+     * Once created, an instance cannot be resized, however it may contain any number of accounts
+     * between zero and its {@link #getCapacity capacity}.
+     *
+     * @param capacity the maximum capacity.
+     *
+     * @throws IllegalArgumentException if capacity is negative.
+     */
+    public Accounts(final int capacity) {
+        super(capacity, Struct.SIZE);
+    }
+
+    Accounts(final ByteBuffer buffer) {
+        super(buffer, Struct.SIZE);
+    }
+
+    /**
+     * Adds a new account at the end of this batch.
+     * <p>
+     * If successfully, moves the current {@link #setPosition position} to the newly created
+     * account.
+     *
+     * @throws IllegalStateException if this batch is read-only.
+     * @throws IndexOutOfBoundsException if exceeds the batch's capacity.
+     */
+    @Override
+    public void add() {
+        super.add();
+    }
+
+    /**
+     * Gets a unique, client-defined identifier for the account.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public byte[] getId() {
+        return getUInt128(at(Struct.Id));
+    }
+
+    /**
+     * Gets a unique, client-defined identifier for the account.
+     *
+     * @param part a {@link UInt128} enum indicating which part of the 128-bit value is to be
+     *        retrieved.
+     * @return a {@code long} representing the the first 8 bytes of the 128-bit value if
+     *         {@link UInt128#LeastSignificant} is informed, or the last 8 bytes if
+     *         {@link UInt128#MostSignificant}.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public long getId(final UInt128 part) {
```

Comment:
I don't understand the point of this function. Why is the user passing anything in here?

Reply:
This method splits a 128-bit value into two 64-bit values representable by a Java long.

It is an important use case when you need this 128-bit value as primitives, and not as a heap-allocated array. 
For example, if you are going to implement your own ID representation.

```java
// Getting a pair of i64
long idLsb = batch.getId(UInt128.LeastSignificant);
long idMsb = batch.getId(UInt128.MostSignificant);

return new MyID(idLsb, idMsb);
```

The corresponding setter does the opposite: `setId(long leastSignificant, long mostSignificant)`.